### PR TITLE
Renovate config fix for AKS GPU CUDA and GRID updates (all types of versions)

### PR DESCRIPTION
**Pull request by ganeshkumarashok**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This fixes the Renovate config so that aks-gpu container images (CUDA and GRID) get automated updates for major, minor, patch, and "pre-release" updates of the container image.
 
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: .github/renovate.json [hunk 1]
```diff
@@ -16,6 +16,17 @@
     }
   ],
   "packageRules": [
+    {
```
Comment (ganeshkumarashok) [inline .github/renovate.json:19]: These changes update the AKS-GPU container images for pre-release/patch versions but don't create PRs for minor version updates. Any suggestions on how to fix that?

Comment (Devinwong) [reply .github/renovate.json:19]: Can you link your fork PR which updates the minor?

Comment (ganeshkumarashok) [reply .github/renovate.json:19]: This is the current fork with the patch version updates: https://github.com/ganeshkumarashok/AgentBaker/blob/master/.github/renovate.json 

For minor versions I just set `enable` to `true` for docker format: https://github.com/ganeshkumarashok/AgentBaker/commit/c29f78597da528b96d04f6c2d58df485481e4a75

Comment (Devinwong) [reply .github/renovate.json:19]: hmmm. No idea yet. If it's still not working, maybe remove "major" and "patch" from "matchUpdateTypes" to see if minor works. Make sure you really have that updated minor version in your container registry? You can search that version in Renovate debug logs directly. If Renovate really finds it, you should be able to see some logs there.

Comment (ganeshkumarashok) [reply .github/renovate.json]: It works in the fork now! Had to move the aks-gpu packageRule to the bottom (as the enablement for minor versions is an exception to the broader rule), and set `ignoreUnstable` to false https://github.com/ganeshkumarashok/AgentBaker/commit/4090b89f581ca03f063dfed8815889113437c5ef

**Review by djsly — APPROVED** [review]

**Pull request merged**